File: apps/ClubHouse-Backend/Models/AppUser.cs
using Microsoft.AspNetCore.Identity;

namespace ClubHouse.Backend.Models
{
    public class AppUser:IdentityUser
    {
        public string Name { get; set; }
        public string Surname { get; set; }
        public DateTime DateOfBirth { get; set; }
        public bool IsMale { get; set; }
        public bool IsBanned { get; set; }
        public double Longitude { get; set; }
        public double Latitude { get; set; }
        public double PrefferedRadiusKm {get;set;}
        public string PrefferedCategories { get; set; }

        public ICollection<ProfileSnapshot> ProfileSnapshots { get; set; }
        public bool AcceptedTermsAndConditions { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime LastUpdatedAt { get; set; }
    }
}

File: apps/ClubHouse-FrontEnd/src/models/AppUser.ts
class AppUser{
     Email:string;
     UserName:string;
     Password:string;
     Name:string;
     Surname:string;
     DOB:Date;
     IsMale:boolean;
     Longitude:number;
     Latitude:number;
     PrefferedRadiusKm:number;
     PrefferedCategories:string;
     AcceptedTermsAndConditions:boolean;

    constructor(
        Email:string,
        UserName:string,
        Password:string,
        Name:string,
        Surname:string,
        DOB:Date,
        IsMale:boolean,
        Longitude:number,
        Latitude:number,
        PrefferedRadiusKm:number,
        PrefferedCategories:string,
        AcceptedTermsAndConditions:boolean,
    ){
        this.Email=Email;
        this.UserName=UserName;
        this.Password=Password;
        this.Name=Name;
        this.Surname=Surname;
        this.DOB=DOB;
        this.IsMale=IsMale;
        this.Longitude=Longitude;
        this.Latitude=Latitude;
        this.PrefferedRadiusKm=PrefferedRadiusKm;
        this.PrefferedCategories=PrefferedCategories;
        this.AcceptedTermsAndConditions=AcceptedTermsAndConditions;
    }

}

export default AppUser;